test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,942 @@
+			}
+
+			if err != nil && testcase.err == "" {
+				t.Logf("actual error: %v", err)
+				t.Logf("expected error: %v", testcase.err)
+				t.Fatal("got error but expected none")
+			}
+
+			if err.Error() != testcase.err {
+				t.Logf("actual validation error: %v", err)
+				t.Logf("expected validation error: %v", testcase.err)
+				t.Error("unexpected validation error")
+			}
+
+			checkFailureReason(t, err, testcase.failureReason)
+		})
+	}
+}
+
+func TestMultiplePolicyBindings(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+	server, err := apiservertesting.StartTestServer(t, nil, nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	paramKind := &admissionregistrationv1alpha1.ParamKind{
+		APIVersion: "v1",
+		Kind:       "ConfigMap",
+	}
+	policy := withPolicyExistsLabels([]string{"paramIdent"}, withParams(paramKind, withPolicyMatch("secrets", withFailurePolicy(admissionregistrationv1alpha1.Fail, makePolicy("test-policy")))))
+	policy.Spec.Validations = []admissionregistrationv1alpha1.Validation{
+		{
+			Expression: "params.data.autofail != 'true' && (params.data.conditional == 'false' || object.metadata.name.startsWith(params.data.check))",
+		},
+	}
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	autoFailParams := makeConfigParams("autofail-params", map[string]string{
+		"autofail": "true",
+	})
+	if _, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), autoFailParams, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+	autofailBinding := withBindingExistsLabels([]string{"autofail-binding-label"}, policy, makeBinding("autofail-binding", "test-policy", "autofail-params"))
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), autofailBinding, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	autoPassParams := makeConfigParams("autopass-params", map[string]string{
+		"autofail":    "false",
+		"conditional": "false",
+	})
+	if _, err := client.CoreV1().ConfigMaps("default").Create(context.TODO(), autoPassParams, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+	autopassBinding := withBindingExistsLabels([]string{"autopass-binding-label"}, policy, makeBinding("autopass-binding", "test-policy", "autopass-params"))
+	if _, err := client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Create(context.TODO(), autopassBinding, metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	condpassParams := makeConfigParams("condpass-params", map[string]string{
+		"autofail":    "false",
+		"conditional": "true",
```

Comment:
I find the combination of these params a bit hard to follow -- wondering if we can make the pass/fail conditions for mulitple bindings more obvious :thinking:

Reply:
Would renaming them help? I'm not sure how much there is to simplify with the logic behind the expression itself (one is directly checked by the expression, the other being a switch that enables another branch of the expression that uses param data).